packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -112,6 +121,25 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ArrowFunctionExpression) {
           // handle arrow function shorthand
 
+          // console.log(invalidAncestor.parent.type === AST_NODE_TYPES.VariableDeclaration)
+
+          if (options.ignoreVoidInVoid) {
+            if (
+              invalidAncestor.returnType &&
+              isVoidLikeType(invalidAncestor.returnType)
+            ) {
+              return;
+            }
+            if (
+              !invalidAncestor.returnType &&
+              invalidAncestor.parent.type ===
+                AST_NODE_TYPES.VariableDeclarator &&
+              isVoidLikeType(invalidAncestor.parent.id.typeAnnotation)
+            ) {
+              return;
+            }
```

Comment:
[Bug] Why do we have it here? We shouldn't take into account this situation (the function is being assigned to a variable with `void` type).

If you added that after my comment  (https://github.com/typescript-eslint/typescript-eslint/pull/8632#discussion_r1519929980), I apologize, it was a typo. 

I was thinking of the following code:

```ts
type Foo = void
const test = (): Foo => console.log('err')
```

Reply:
thank you! delete the code and test code.